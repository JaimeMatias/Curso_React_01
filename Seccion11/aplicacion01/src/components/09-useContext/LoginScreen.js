import React, { useContext } from 'react'

import { UserContext } from './UserContext'

export const LoginScreen = () => {
  const { user, setuser } = useContext(UserContext);
  console.log(user)
  const nombre = {
    nombre: "Matias",
    correo: "matias@gmial",
    id: 123
  }
  const handleClick = () => {
    setuser(nombre)
  }
  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button className='btn btn-primary' onClick={handleClick}>LogIn</button>
    </div>
  )
}

