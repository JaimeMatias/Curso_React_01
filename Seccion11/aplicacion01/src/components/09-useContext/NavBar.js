import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-light bg-light">
            <NavLink exact="true"className="navbar-brand" to="/">UseContext</NavLink>

            <div className="collapse navbar-collapse" id="navbarNav">
                <ul className="navbar-nav">
                    <li className="nav-item">
                        <NavLink exact="true"  className="nav-link" aria-current="page" to="/">Home</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink exact="true"  className="nav-link" to="/about">About</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink exact="true" className="nav-link" to="/login">Login</NavLink>
                    </li>

                </ul>
            </div>
        </nav>

    )
}
