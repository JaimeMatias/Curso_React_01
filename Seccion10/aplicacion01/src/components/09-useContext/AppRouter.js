import React from 'react'
import {
    BrowserRouter as Router,
    Routes,
    Route,
    Link,
    Redirect
} from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';


export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <Routes>

                    <Route exact path="/about" element={<AboutScreen />} />
                    <Route exact path="/login" element={<LoginScreen />} />
                    <Route exact path="/" element={<HomeScreen />} /> {/*La ruta mas general debe estar siempre abajo*/}
                    <Route path="*" element={<HomeScreen />} /> {/*Cualquier ruta que no coincida con las anteirores me la panda al HomeScreens*/}

                </Routes>

            </div>
        </Router>

    )
}


