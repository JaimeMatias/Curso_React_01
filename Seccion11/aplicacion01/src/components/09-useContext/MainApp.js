import React, { useState } from 'react'
import { AboutScreen } from './AboutScreen'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'


export const MainApp = () => {
const [user,setuser]=useState({})
  
  return (
    <div>
      <UserContext.Provider value={{user,setuser}}>

        <AppRouter />
      </UserContext.Provider>


    </div>
  )
}

