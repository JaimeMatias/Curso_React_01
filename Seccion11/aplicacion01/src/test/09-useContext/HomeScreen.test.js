import { mount, shallow } from 'enzyme';
import React from 'react';
import { HomeScreen } from '../../components/09-useContext/HomeScreen';
import { UserContext } from '../../components/09-useContext/UserContext';

describe('Prueba del <Home Screen> ', () => {
    const user={
        name:'Matias',
        email:'matias@gmail.com'
    }
    const wrapper = mount(
        <UserContext.Provider value={{user}}>
            <HomeScreen />
        </UserContext.Provider>

    );

    test('debe de mostrarse corretamente', () => {

        expect(wrapper).toMatchSnapshot();
    })

})
