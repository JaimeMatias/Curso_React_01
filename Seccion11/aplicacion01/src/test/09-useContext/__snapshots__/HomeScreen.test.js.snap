// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del <Home Screen>  debe de mostrarse corretamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Matias",
   "email": "matias@gmail.com"
}
  </pre>
</div>
`;
